### PR TITLE
Allow mapping scalars to simple generic types

### PR DESCRIPTION
**Pull request by BoD**
See #6156 and https://github.com/apollographql/apollo-kotlin/issues/3243. This is far from perfect but unlocks a few cases.

### CODE REVIEW

File: libraries/apollo-compiler/src/test/kotlin/com/apollographql/apollo/compiler/CodegenTest.kt [hunk 1]
```diff
@@ -304,13 +305,15 @@ class CodegenTest {
...
               "String" to ScalarInfo("java.lang.String", ExpressionAdapterInitializer("new com.example.MyStringAdapter()")),
+              "ListOfString" to ScalarInfo("java.util.List<java.lang.String>"),
           )
         } else {
           mapOf(
               "Date" to ScalarInfo("java.util.Date"),
               "URL" to ScalarInfo("kotlin.String", ExpressionAdapterInitializer("com.example.UrlAdapter")),
               "ID" to ScalarInfo("kotlin.Long"),
               "String" to ScalarInfo("kotlin.String", ExpressionAdapterInitializer("com.example.MyStringAdapter()")),
+              "ListOfString" to ScalarInfo("kotlin.collections.List<kotlin.String?>"),
```
Comment (martinbonnin) [inline libraries/apollo-compiler/src/test/kotlin/com/apollographql/apollo/compiler/CodegenTest.kt:316]: That is quite the mouthful. I guess `List<String?>` doesn't work here?

**Review by martinbonnin — APPROVED** [review]
LGTM. 

2 possible improvements:

1. Make it a proper parser for Kotlin [`type`](https://kotlinlang.org/spec/syntax-and-grammar.html#grammar-rule-type) (unlocks `dynamic`, function types, variance, wildcards, type annotations, etc...)
2. recognize some frequent auto imports (`kotlin.`, `kotlin.collections.`)

File: libraries/apollo-compiler/src/test/kotlin/com/apollographql/apollo/compiler/CodegenTest.kt (hunk 1 above)
Comment (BoD) [reply libraries/apollo-compiler/src/test/kotlin/com/apollographql/apollo/compiler/CodegenTest.kt:316]: Just tried to make sure, but no, `ClassName.bestGuess()` expects a fully qualified name. I'll see if I can add a few shortcuts for `String`, `List`, etc.

File: libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/codegen/kotlin/helpers/bestGuess.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+private fun String.withPackage(): String {
+  return if (this in commonKotlinTypes) {
+    "kotlin.${this}"
+  } else if (this in commonKotlinCollectionsTypes) {
+    "kotlin.collections.${this}"
+  } else {
+    this
+  }
+}
```
Comment (martinbonnin) [inline libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/codegen/kotlin/helpers/bestGuess.kt:34]: 👍

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**